lib/py/tests/edge_context_tests.py

```diff
@@ -359,3 +365,5 @@ def test_service_auth_with_additional_options(self):
         self.assertEqual(request_context.service.on_behalf_of_id, "t2_deadbeef")
         self.assertEqual(request_context.service.on_behalf_of_roles, ["admin"])
         self.assertEqual(request_context.service.requests_elevated_access, True)
+        with self.assertRaises(NoAuthenticationError):
```

Comment:
why would this case raise `NoAuthenticationError`? this should be a properly service auth'd case?

Reply:
if it's service auth, then attempting to access user auth should fail (hence `request_context.user.id`)